runtime/vm.go

```diff
@@ -109,13 +108,25 @@ func (v *VM) Run() (err error) {
 }
 
 func (v *VM) run() {
-	for v.ip < v.curIPLimit && (atomic.LoadInt64(&v.aborting) == 0) {
+	defer func() {
+		if r := recover(); r != nil {
+			if v.ip < v.curIPLimit {
```

Comment:
If we keep this panic-recover approach, we can replace this to `if v.ip < len(v.curInsts)-1` and remove `v.curIPLimit` entirely,